proxy/src/control/destination/mod.rs

```diff
@@ -240,13 +243,14 @@ impl Metadata {
         Metadata {
             dst_labels: None,
             // If we have no metadata on an endpoint, assume it does not support TLS.
-            tls_identity: None,
+            tls_identity:
+                Conditional::None(tls::ReasonForNoIdentity::NotProvidedByServiceDiscovery),
```

Comment:
tioli: consider implementing `impl T From<tls::ReasonForNoIdentity> for Conditional<T, tls::ReasonForNoIdenitty>> { ... }`

This also makes me think that we might consider more narrowly scoping `Conditional` as `ConditionalTls` by specializing the `None` type with `tls::ReasonForNoIdentity` (that is, if there are no other None types).It appears this could reduce a bit of boilerplate.

Reply:
Later we'll have more reasons for why there's no configuration available.